### PR TITLE
Move ERC20 collateral tests to new style, port moveCollateral require…

### PR DESCRIPTION
**Pull request by grandizzy**
…s to custom errors

### CODE REVIEW

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol [hunk 1]
```diff
@@ -139,13 +166,52 @@ abstract contract ERC20HelperContract is ERC20DSTestPlus {
 
     }
 
-    function _addLiquidity(AddLiquidity memory specs_) internal {
+    function _addLiquidity(AddLiquiditySpecs memory specs_) internal {
         changePrank(specs_.from);
         for (uint256 i = 0; i < specs_.amounts.length; ++i) {
             _pool.addQuoteToken(specs_.amounts[i].amount, specs_.amounts[i].index);
```
Comment (MikeHathaway) [inline src/_test/ERC20Pool/ERC20DSTestPlus.sol:172]: should we also check the `AddQuoteToken()` emits here like the other helper methods?

File: src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol [hunk 2]
```diff
@@ -36,112 +36,146 @@ contract ERC20ScaledCollateralTest is ERC20HelperContract {
...
-        _pool.addQuoteToken(10_000 * 1e18, depositIndexHigh);
-        _pool.addQuoteToken(10_000 * 1e18, depositIndexMed);
-
-        // check initial pool state
-        assertEq(_pool.htp(), 0);
-        assertEq(_pool.lup(), BucketMath.MAX_PRICE);
-
-        assertEq(_pool.poolSize(),     30_000 * 1e18);
-        assertEq(_pool.borrowerDebt(), 0);
+        Liquidity[] memory amounts = new Liquidity[](5);
```
Comment (MikeHathaway) [inline src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol]: array is larger than it needs to be here

File: src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol [hunk 3]
```diff
@@ -36,112 +36,146 @@ contract ERC20ScaledCollateralTest is ERC20HelperContract {
...
+                indexLimit:   3_000,
+                price:        2_981.007422784467321543 * 1e18
+            })
+        );
 
-        // check borrower state
-        (uint256 borrowerDebt, , uint256 borrowerCollateral, ) = _pool.borrowerInfo(_borrower);
-        assertEq(borrowerDebt,       _pool.borrowerDebt());
-        assertEq(borrowerCollateral, _pool.pledgedCollateral());
-        assertEq(
```
Comment (MikeHathaway) [inline src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol:90]: Some of these checks might still be valuable - i.e check total borrower and pool debt is the same. That was one of the ways that we identified dust issues in the v8.1 implementation

File: src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol [hunk 4]
```diff
@@ -36,112 +36,146 @@ contract ERC20ScaledCollateralTest is ERC20HelperContract {
...
+                debt:              21_020.192307692307702000 * 1e18,
+                pendingDebt:       21_020.192307692307702000 * 1e18,
+                collateral:        100 * 1e18,
+                collateralization: 14.181637252165253251 * 1e18,
+                inflator:          1 * 1e18
+            })
         );
         assertEq(_collateral.balanceOf(_borrower), 50 * 1e18);
 
-        assertEq(_pool.borrowerCollateralization(borrowerDebt, borrowerCollateral, _pool.lup()), _pool.poolCollateralization());
```
Comment (MikeHathaway) [inline src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol:96]: Same here - may want to ensure pool and borrower state stays in sync

File: src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol (hunk 3 above)
Comment (MikeHathaway) [reply src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol:90]: Could maybe add a `_checkPoolBorrowerInvariants` helper method or the like?

File: src/_test/ERC20Pool/ERC20DSTestPlus.sol (hunk 1 above)
Comment (grandizzy) [reply src/_test/ERC20Pool/ERC20DSTestPlus.sol:172]: yep, plan is to do when get to quote token tests, so far they weren't checked in tests

File: src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol (hunk 2 above)
Comment (grandizzy) [reply src/_test/ERC20Pool/ERC20ScaledPoolCollateral.t.sol]: changes with https://github.com/ajna-finance/contracts/pull/249/commits/92c517d8e4a8ed8e5b766e6161760a23a90099d3

**Comment by grandizzy** [comment]
@MikeHathaway I created a card to track adding invariants test, will do events check for add liquidity when porting quote token related tests

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**